### PR TITLE
feat: Selectable db query performance logging

### LINKED ISSUE #385: 🛠️ [TASK] : Selectable DB query performance logging.
### Summary

We need a way to profile DB queries in a running cat-gateway server.

### Description

Postgres has the ability to display detailed performance statistics and query optimization data for a query when it runs.
We need a way to toggle that dynamically in any running cat-gateway instance.

The task is to.
1. Add an endpoint, protected by a ENV var set "API key".
    a. This endpoint will take a set of flags to enable.  but for now we have 1 flag.  "Enable deep query inspection."
2. When that flag is set `true` every query that is sent to the DB by cat-gateway needs to be prepended with `EXPLAIN ANALYZE`
    a. The query itself is then logged at DEBUG level
    b. The result of the analysis is also logged at debug level.
    c. Make sure there is some kind of identifier attached to the query and the result so that we can piece together interleaved results from multiple queries.  An easy way to do that is make a random uuid when this option is enabled, and add it to the log of the query and its result.
    
We need this fast, so implement it quick and dirty to begin (don;t worry about API key, or generality, focus on just getting the logging working and a endpoint to toggle it on/off.

After we deliver the first q&D version, we need to clean it up so its easy to use consistently and reliably.
In cleaning it up, it would be good to add a category, so when we toggle it on we can toggle it on globally, or for "this set" of queries.  Getting that right will take longer and is not necessary for this immediate task.

See:
* https://www.crunchydata.com/blog/get-started-with-explain-analyze 
* https://www.postgresql.org/docs/current/using-explain.html#USING-EXPLAIN-ANALYZE

for extra details on how this will work.
Note: We do NOT want the query results themselves to be logged, just the result of the query analysis AFTER it has run.

### PR DESCRIPTION
**Pull request by saibatizoku**
# Description

Adds `GET /health/inspection` endpoint to toggle deep query inspection and set log level filtering at runtime.

## Related Issue(s)

List the issue numbers related to this pull request.

Closes #385 

## Description of Changes

1. This endpoint will take a set of flags to enable. but for now we have 1 flag. "Enable deep query inspection."
When that flag is set true every query that is sent to the DB by cat-gateway needs to be prepended with EXPLAIN ANALYZE
2. The query itself is then logged at DEBUG level
3. The result of the analysis is also logged at debug level.
4. Make sure there is some kind of identifier attached to the query and the result so that we can piece together interleaved results from multiple queries. An easy way to do that is make a random uuid when this option is enabled, and add it to the log of the query and its result.

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [ ] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

**Comment by stevenj** [comment]
The changes you have made look OK so far, but I do think we need a test for "SELECT" or "RETURNING" appearing in a modify operation, or NOT appearing in a query.  And then throw an error if the query is wrong.
We also then need to apply these new functions to all the places `query` is used.

Thats why I suggested maybe it would be better to just keep `query` and `query_one` as is, and have logic like this in it:
```
if deep_inspection then
  if query_string has "SELECT" or "RETURNING" in it, then
     plan query with rollback
     execute query
  else
     plan query without rollback
  end
else
  exectue query
end
```
(I am sure you can do better logic than that, its just illustrative).
I would use the stringzilla library to do that substring search, as its the fastest way and we want to minimize overhead here.
This way its easier to call, and we can't get it wrong.
It also simplifies the implementation.
We can also only do the "Does the query have SELECT or RETURNING" when deep inspection is enabled, so in the normal case, there is no overhead to running the query.

What do you think?

**Comment by saibatizoku** [comment]
> What do you think?

It sounds like a plan, I'd only make a distinction for sub-select statements in operations like UPDATE, etc., which won't return and which will return positive on a substring search of "SELECT".

File: catalyst-gateway/bin/src/event_db/mod.rs [hunk 1]
```diff
@@ -211,3 +212,36 @@ pub(crate) async fn establish_connection(url: Option<String>) -> anyhow::Result<
...
 }
+
+/// Determine if the statement is a query statement.
+///
+/// If the query statement starts with `SELECT` or contains `RETURNING`, then it is a
+/// query.
+#[allow(dead_code)]
+fn is_query_stmt(stmt: &str) -> bool {
+    matches!(
+        (stmt.sz_find("SELECT"), stmt.sz_find("RETURNING"),),
```
[the remark this answers is not in the document]
Comment (stevenj) [reply catalyst-gateway/bin/src/event_db/mod.rs]: The changes look good.

**Comment by saibatizoku** [comment]
@stevenj This should be ready to go, it just needs your say so.

**Review by Mr-Leshiy — APPROVED** [review]
LGTM

**Review by stevenj — APPROVED** [review]

**Pull request merged**